src/main/java/edu/kis/powp/jobs2d/drivers/MacroDriver.java

```diff
@@ -0,0 +1,47 @@
+package edu.kis.powp.jobs2d.drivers;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.command.DriverCommand;
+import edu.kis.powp.jobs2d.command.OperateToCommand;
+import edu.kis.powp.jobs2d.command.SetPositionCommand;
+
+
+
+
+public class MacroDriver implements Job2dDriver {
+
+    List<DriverCommand> commands = new ArrayList<DriverCommand>();
+    private Job2dDriver jd;
+    public void clearMacro(){
+        commands.clear();
+    }
+    public void setPreviousDriver(Job2dDriver jd){
```

Comment:
In decorator pattern this could be considered a CORE driver (setCoreDriver) or simply wrapped (or decorated) driver.